simphony/testing/abc_check_engine.py

```diff
@@ -177,15 +177,17 @@ def test_delete_dataset(self):
             name = "test_" + str(i)
             engine.add_dataset(self.create_dataset(name=name))
 
+        datasets_and_names = [(ds, ds.name) for ds in engine.iter_datasets()]
+
         # delete each of the datasets
-        for ds in engine.iter_datasets():
-            engine.remove_dataset(ds.name)
+        for (dataset, name) in datasets_and_names:
+            engine.remove_dataset(name)
             # test that we can't get deleted datasets
             with self.assertRaises(ValueError):
-                engine.get_dataset(ds.name)
+                engine.get_dataset(name)
             # test that we can't use the deleted datasets
             with self.assertRaises(Exception):
-                self.compare_dataset(ds, ds)
+                self.compare_dataset(dataset, dataset)
```

Comment:
This line still causes an error, because ```dataset``` is referencing to the proxy object, and therefore the proxy object is not deleted.

I don't see a clean way how this could be fixed, other than removing this part of the test completely. In Python the object is deleted when there is no references to it, and even trying this test here requires that there is a reference existing.

Maybe compare_dataset could be modified to check if the proxy object has a ```_am_i_deleted``` attribute set to True, and in that case return an exception, but this doesn't sound a very elegant solution.